src/deps/libuwsockets.cpp

```diff
@@ -1577,4 +1592,25 @@ extern "C"
     s->context->loop->data.last_write_failed = 1;
     us_poll_change(&s->p, s->context->loop, LIBUS_SOCKET_READABLE | LIBUS_SOCKET_WRITABLE);
   }
+
+
+  uint64_t uws_res_get_remote_address_info(uws_res_t *res, const char **dest, int *port, bool *is_ipv6)
+  {
+      static thread_local char b[64];
+      auto length = us_get_remote_address_info(b, (us_socket_t *)res, dest, port, (int*)is_ipv6);
+
+      if (length == 4) {
+          length = sprintf(b, "%u.%u.%u.%u", b[0], b[1], b[2], b[3]);
+          *is_ipv6 = false;
+      } else {
+          length = sprintf(b, "%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x",
+              b[0], b[1], b[2], b[3], b[4], b[5], b[6], b[7], b[8], b[9], b[10], b[11],
+              b[12], b[13], b[14], b[15]);
+          *is_ipv6 = true;
```

Comment:
this IPv6 formatter was taken from some usockets code. Maybe this can be later improved to a more intelligent IPv6 formatter. I don't know enough about ipv6 formatting to do this though.

Reply:
Maybe some utils like `inet_ntop`.

```zig
if (length == @sizeOf(std.meta.FieldType(std.os.sockaddr.in6, .addr))) {
    _ = bun.c_ares.ares_inet_ntop(std.os.AF.INET6, &b, &new_buf, new_buf_len);
    const len_ = bun.len(bun.cast([*:0]u8, new_buf[0..]));
    const ip = bun.String.create(new_buf[0..len_]);
}
```